code/obj/item/gun/kinetic.dm

```diff
@@ -870,6 +880,267 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 			projectiles = list(current_projectile)
 			UpdateIcon()
 
+/obj/item/gun/kinetic/uzi
+	desc = "A stamped metal PDW."
+	name = "\improper MOR-30"
+	icon_state = "uzi"
+	item_state = "uzi"
+	spread_angle = 8
+	shoot_delay = 5
+	has_empty_state = TRUE
+	w_class = W_CLASS_SMALL
+	force = MELEE_DMG_PISTOL
+	ammo_cats = list(AMMO_PISTOL_9MM_ALL)
+	max_ammo_capacity = 30
+	auto_eject = TRUE
+	fire_animation = TRUE
+	default_magazine = /obj/item/ammo/bullets/nine_mm_surplus/mag_mor
+
+	get_desc(dist, mob/user)
+		if (user.get_gang() != null)
+			. += "For when you need MOR' DAKKA. Uses 9mm NATO rounds."
+		else
+			. += "Its' firemodes are labelled 'DAKKA' and 'MOR'... Uses 9mm NATO rounds."
+
+	New()
+		ammo = new default_magazine
+
+		set_current_projectile(new/datum/projectile/bullet/nine_mm_surplus/burst)
+		projectiles = list(current_projectile, new/datum/projectile/bullet/nine_mm_surplus/auto)
+		AddComponent(/datum/component/holdertargeting/fullauto, 1.5, 1.5, 1)
+		..()
+
+	attack_self(mob/user)
+		..()	//burst shot has a slight spread.
+		if (istype(current_projectile, /datum/projectile/bullet/nine_mm_surplus/auto))
+			spread_angle = 10
+			shoot_delay = 4
+		else
+			spread_angle = 8
+			shoot_delay = 5
+
+
+/obj/item/gun/kinetic/greasegun
+	name = "\improper Grease Gun"
+	desc = "A stamped metal SMG, cheaply produced years ago, and found in the weirdest places today."
+	icon_state = "grease"
+	item_state = "grease"
+	icon = 'icons/obj/items/guns/kinetic48x32.dmi'
+	spread_angle = 8
+	shoot_delay = 5
+	has_empty_state = TRUE
+	w_class = W_CLASS_SMALL
+	force = MELEE_DMG_PISTOL
+	ammo_cats = list(AMMO_PISTOL_9MM_ALL)
+	max_ammo_capacity = 30
+	auto_eject = TRUE
+	fire_animation = TRUE
+	default_magazine = /obj/item/ammo/bullets/nine_mm_surplus/mag_grease
+	var/greased = FALSE //guh
+	var/glued = FALSE // :'(
+	get_desc(dist, mob/user)
+		if (!greased && !glued)
+			. += "It's all seized up and could do with maintenance."
+		else if (glued)
+			. += "It's, er, all sticky and covered glue. WHY is it covered with glue???"
+		else
+			. += "It's greasy, alright..."
+
+	attack_self(mob/user as mob)
+		if(ishuman(user))
+			if(two_handed)
+				setTwoHanded(0) //Go 1-handed.
+				src.spread_angle = initial(src.spread_angle)
+				icon_state = "grease"
+			else
+				if(!setTwoHanded(1)) //Go 2-handed.
+					boutput(user, SPAN_ALERT("Can't switch to 2-handed while your other hand is full."))
+				else
+					icon_state = "greaseunfolded"
+					src.spread_angle = 4
+		..()
+
+	shoot() // fuck up firerate speed
```

Comment:
Should maybe get ungreased again with use?

Reply:
hmmm. yes... that could make this even more Interesting